File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    time = math.ceil((100 - progresses[0]) / speeds[0])
    deployed_tasks = 0
    
    for progress, speed in zip(progresses, speeds):
        current_progress = progress + speed * time
        
        if current_progress >= 100:
            deployed_tasks += 1
        else:
            time += math.ceil((100 - current_progress) / speed)
            answer.append(deployed_tasks)
            deployed_tasks = 1
    
    if deployed_tasks:
        answer.append(deployed_tasks)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
  const releaseCounts = [];
  const requiredDays = progresses.map((progress, idx) =>
    Math.ceil((100 - progress) / speeds[idx])
  );
  let maxDays = requiredDays[0];
  let completedFeatureCount = 1;

  for (let i = 1; i < requiredDays.length; i++) {
    if (requiredDays[i] <= maxDays) completedFeatureCount++;
    else {
      releaseCounts.push(completedFeatureCount);
      maxDays = requiredDays[i];
      completedFeatureCount = 1;
    }
  }
  releaseCounts.push(completedFeatureCount);

  return releaseCounts;
}
